### ISSUE TITLE
Share terminal access during collaborative sessions

### ISSUE THREAD

**Issue by fdionisi** [zed-industries/zed#4767]
### Check for existing issues

- [X] Completed

### Describe the feature

It would be handy to allow sharing of Zed terminals during collaborative sessions. Allowing users to interact with a number of terminal views (perhaps allowing to set read/write permissions) would make the collaborative experience impeccable.

### If applicable, add mockups / screenshots to help present your vision of the feature

No screenshot and mocks, but here's [VS Code documentation](https://learn.microsoft.com/en-us/visualstudio/liveshare/use/share-server-visual-studio-code#share-a-terminal) as a reference.

**Comment by joefitzgerald** [comment]
Now that terminal sharing is possible via SSH remoting, what is the plan to enable this for collaboration sessions that are not using SSH? For now we're using `tmate` to handle this, but it would be great to be able to stay inside of Zed.

**Comment by SomeoneToIgnore** [comment]
> Now that terminal sharing is possible via SSH remoting

It's not really true, Zed merely does `ssh -c ....` on the ssh client side, so no real sharing happens.
Nothing collaborative is possible still in Zed's terminal.

**Comment by joefitzgerald** [comment]
I must have misread this from the [SSH Remoting Is Here!](https://zed.dev/blog/remote-development) blog post then? I haven't actually used the feature yet, and assumed that SSH remoting support enabled the terminal pane to spawn new terminals on te remote host.

> This means that you can open new terminals and spawn tasks without having to retype your passphrase or re-authenticate.

**Comment by joefitzgerald** [comment]
Any updates on where this might fit in the overall roadmap?

**Comment by h3h** [comment]
+1 for this being a major missing piece for mobbing code in an edit, test (in terminal), repeat loop. We resort to screen sharing, which distracts from the shared Zed editor panes and sort of defeats the purpose of collaborative mode.

**Comment by blaykelarue** [comment]
+1 !

**Comment by videni** [comment]
Yes, the terminal is really important to our development workflow. Without it, the collaboration feature would be useless for me. It seems like most users never use Zed's collaboration feature, but this one is truly a game-changer.

**Comment by 0xBA5E64** [comment]
I would like to see clarification on how exactly Zed's collaboration feature is even _meant_ to be used right now without shared terminal access. Thus far, I don't really understand how something as simple as *~~a language server would work during collaboration, let alone~~*\* shared collaborative debugging of a program, without at least a shared terminal. 

\*Language server forwarding not working turned out to be mostly a momentary issue, good on the zed team for implementing that! -even if it's in complete isolation to all other service forwarding.  

I could see how, if collaboration mode instead just cloned and synced your colleague's project directory somewhere local on your system, and you were able to run it and debug it yourself, you'd at least perhaps be able to do rudimentary debugging on each end separately. As of right now, I don't really understand how the collaboration feature is supposed to be used as anything more advanced than a "dumb" multiplayer notepad, as opposed to a proper shared development environment.

The fact the Terminal seems _fully_ disabled when joining collaboration sessions also means one can't for instance set up and SSH into a shared tmux session on the collaborator's host machine, unless entirely external to Zed, with an external app.
